src/Nethermind/Nethermind.Trie/PatriciaTree.cs

```diff
@@ -319,7 +319,7 @@ private void SetRootHash(Keccak? value, bool resetObjects)
             }
             catch (TrieException e)
             {
-                throw new TrieException($"Failed to load key {rawKey.ToHexString()} from root hash {rootHash ?? RootHash}.", e);
+                throw TrieException.CreateOnLoadFailure(rawKey, rootHash ?? RootHash, e);
```

Comment:
Change to void method and put the throw in there (throw is chunky)

Then on the method add the attributes
```
[DoesNotReturn]
[StackTraceHidden]
```
`DoesNotReturn` tells the Jit it doesn't need to add clean up code after the call and can just put a `int3` after it.

`StackTraceHidden` says to exclude the method from the stack trace so it looks like it was thrown directly in the caller, like the throw was still there

Reply:
e.g. `CreateOnLoadFailure` => `ThrowOnLoadFailure`

Reply:
Done